File: TF13. LISTS/P06.SquareNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace P06.SquareNumbers
{
    class Program
    {
        static void Main(string[] args)
        {
            List<int> input = Console.ReadLine()
                        .Split()
                        .Select(int.Parse)
                        .ToList();

            List<int> result = new List<int>();

            foreach (int digit in input)
            {
                if (Math.Sqrt(digit) == (int)Math.Sqrt(digit))
                {
                    result.Add(digit);
                }
            }

            result.Sort();
            result.Reverse();

            Console.WriteLine(string.Join(" ", result));
        }
    }
}

File: TF13. LISTS (Java)/P06_SquareNumbers.java
import java.util.*;
import java.util.stream.Collectors;

public class P06_SquareNumbers {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);

        List<Integer> input = Arrays.stream(in.nextLine().split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());

        List<Integer> result= new ArrayList<>();

        for (int digit: input) {

            if (Math.sqrt(digit) == (int)Math.sqrt(digit)) {

                result.add(digit);
            }
        }

        Collections.sort(result);
        Collections.reverse(result);

        for (int print: result) {

            System.out.printf("%d ", print);
        }
    }
}
